### PR TITLE
Closes #2 Implement Clone and adds len() to MmapReadOnly

### LINKED ISSUE #2: Fuzzy prefix searches?
Wow, what a slick library. I was just evaluating this for use in building custom indices, and I'm really impressed. I think I'm going to get a lot of mileage out of this. :-)

However, I ran into one use case that would be very handy: Fuzzy prefix matching. Given a string `Masach`, I can search for `Levenshtein::new("Masach", 1)` or `Regex::new("Masach.*")`. But what I'd really love to do is autocomplete to `Massachusetts`, with the extra `s`.

It seems like it should be possible to build something like:

```rust
// Match anything with a prefix that would be matched by A.
impl<A: Automaton> Automaton for Prefix<A> { ... }
```

But I'm not quite sure how to go about something like that. Is this basically trivial, or is it much harder than it looks?

### PR DESCRIPTION
**Pull request by fulmicoton**

### CODE REVIEW

File: src/raw/mmap.rs [hunk 1]
```diff
@@ -50,6 +57,23 @@ impl MmapReadOnly {
...
+
+    fn clone(&self) -> MmapReadOnly {
+        MmapReadOnly{
+            map: self.map.clone(),
+            offset: self.offset,
+            len: self.len,
+        }
+    }
+
+    fn clone_from(&mut self, source: &Self) {
```
Comment (BurntSushi) [inline src/raw/mmap.rs]: I think we can leave this specialization out. The default implementation of `clone_from` should be good enough. :-)

**Comment by BurntSushi** [comment]
@fulmicoton One nit, but otherwise looks good! Thanks!

**Comment by fulmicoton** [comment]
Amended the commit accordingly.

**Comment by BurntSushi** [comment]
Fantastic, thanks!

**Comment by BurntSushi** [comment]
Released on crates.io in `0.1.27`.

**Comment by fulmicoton** [comment]
Awesome thanks.

**Pull request merged**